devlab/tests/test_verify_dst_functionality.py

```diff
@@ -0,0 +1,378 @@
+
+        # declare vars in case if condition below will not be satisfied
+        image_name = ''
+        nic_name = ''
+        self.neutron_float_ip_quota = 0
+        # create vm parameters
+        flavor_name = config.flavors[0]['name']
+        for tenant in config.tenants:
+            if tenant['name'] == TEST_TENANT_NAME:
+                image_name = tenant['images'][0]['name']
+                nic_name = tenant['networks'][0]['name']
+                self.neutron_float_ip_quota = \
+                    tenant['neutron_quotas'][0]['floatingip']
+
+        # fetch public_key and according keypair from config
+        public_key_name = [x['name'] for x in config.keypairs
+                           if x['name'] == config.private_key['name']][0]
+        key_pair = [keypair for keypair in config.keypairs
+                    if keypair['name'] == public_key_name][0]
+
+        # delete user from key_pair
+        if 'user' in key_pair:
+            del key_pair['user']
+
+        self.release_fips_tenant()
+
+        self.external_networks_ids_list = \
+            [i['id'] for i in self.dst_cloud.neutronclient.list_networks(
+            )['networks'] if i['router:external'] is True]
+
+        fip = self.dst_cloud.neutronclient.create_floatingip(
+            {"floatingip": {"floating_network_id":
+                            self.external_networks_ids_list[0]}})
+        pprint.pprint(fip)
+        fip = self.dst_cloud.neutronclient.create_floatingip(
+            {"floatingip": {"floating_network_id":
+                            self.external_networks_ids_list[0]}})
+
+        self.float_ip_address = fip['floatingip']['floating_ip_address']
+        self.float_ip_id = fip['floatingip']['id']
+
+        self.fip_quota_neutron = self.dst_cloud.neutronclient.show_quota(
+            self.dst_tenant_id)['quota']['floatingip']
+
+        # get keypair quota for tenant
+        self.keypair_quota = self.dst_cloud.novaclient.quotas.get(
+            tenant_id=self.dst_tenant_id).key_pairs
+        # delete tenant keys if there are any
+        if self.dst_cloud.novaclient.keypairs.list():
+            for key_p in self.dst_cloud.novaclient.keypairs.list():
+                key_p.delete()
+        # create new keypair
+        self.dst_cloud.novaclient.keypairs.create(**key_pair)
+
+        sec_group = self.create_security_group_and_rules(SECURITY_GROUP_NAME)
+        self.TST_IMAGE = {'image': self.dst_cloud.get_image_id(image_name),
+                          'flavor': self.dst_cloud.get_flavor_id(flavor_name),
+                          'nics': [{"net-id": self.dst_cloud.get_net_id(
+                              nic_name)}],
+                          'name': TEST_VM_NAME + str(time.ctime()).replace(
+                              ' ', '').replace(':', '_'),
+                          'key_name': public_key_name,
+                          'security_groups': [sec_group['name']]}
+
+    def tearDown(self):
```

Comment:
Modify the logic to:
1. Remove all dependencies
1.1 Detach volumes
1.2 Delete volumes
2. Remove VMs
3. Remove FIPs